data-science-onramp/data-cleaning/clean.py

```diff
@@ -129,26 +137,33 @@ def compute_end_udf(duration, start, end):
     return compute_time(duration, start, end)[2]
 
 
+# [END datascienceonramp_timehelperudf]
+
+# [START datascienceonramp_sparksession]
 if __name__ == "__main__":
-    TABLE = sys.argv[1]
-    BUCKET_NAME = sys.argv[2]
+    BUCKET_NAME = sys.argv[1]
+    TABLE = sys.argv[2]
 
     # Create a SparkSession, viewable via the Spark UI
     spark = SparkSession.builder.appName("data_cleaning").getOrCreate()
 
-    # Load data into dataframe if table exists
+    # Load data into temporary dataframe if table exists, drop unused column and load into new dataframe
+    # (using a temporary dataframe is an alternative to calling df.drop(column_name).collect() and is used
+    # to avoid OOM issues)
     try:
-        df = spark.read.format("bigquery").option("table", TABLE).load()
+        temp_df = spark.read.format("bigquery").option("table", TABLE).load()
+        df = temp_df.drop("gender")
```

Comment:
Should this be more explicitly explained in the code why we are removing gender? Either way, I would move it outside of this try-block. It's also ok to do `df = df.<operation>` and not use the `temp_df` var name as this ends up happening in multiple other places in the script.

Reply:
Np for moving it outside of try block and making that df = df.operation change. Reasoning for removal of the column is in the **Putting it together** section of the doc I sent you - do you think it also needs to be mentioned here?

Reply:
I think that's fair, maybe it should have its own region tag then?

Reply:
Added region tag, but have to wait a few min before re-staging